<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ur5e_robot.urdf.xacro          | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="UR5e_robotiq" xmlns:xacro="http://ros.org/wiki/xacro">
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>
-->
  </gazebo>
  <link name="table">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.0074245 0.051498 0.61219"/>
      <mass value="30.118"/>
      <inertia ixx="5.2287" ixy="0.0061081" ixz="0.022985" iyy="4.6538" iyz="-0.29996" izz="5.6994"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://catch_control/urdf/table.STL"/>
      </geometry>
      <material name="">
        <color rgba="0.64706 0.61961 0.58824 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://catch_control/urdf/table.STL"/>
      </geometry>
    </collision>
  </link>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/</robotNamespace>
    </plugin>
  </gazebo>
  <!--<link name="world"/>-->
  <!--<joint name="world_fixed" type="fixed">-->
  <!--<origin xyz="0 0 0" rpy="0 0 0"/>-->
  <!--<parent link="world"/>-->
  <!--<child link="gripper_root_link"/>-->
  <!--</joint>-->
  <link name="gripper_root_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0  0  0"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1e2"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000001" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000001" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000001"/>
    </inertial>
  </link>
  <joint name="gripper_base_joint" type="fixed">
    <parent link="gripper_root_link"/>
    <child link="gripper_base_link"/>
    <origin rpy="0 0 0" xyz="0.0 0 0"/>
  </joint>
  <link name="gripper_base_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_base_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_base_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.636951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000380" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.001110" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.001171"/>
    </inertial>
  </link>
  <gazebo reference="gripper_base_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <joint name="gripper_finger1_joint" type="revolute">
    <parent link="gripper_base_link"/>
    <child link="gripper_finger1_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0.0 0.0" xyz="0.05490451627 0.03060114443 0.0"/>
    <limit effort="1000" lower="0.0" upper="0.804" velocity="2.0"/>
  </joint>
  <joint name="gripper_finger2_joint" type="revolute">
    <parent link="gripper_base_link"/>
    <child link="gripper_finger2_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.05490451627 -0.03060114443 0.0"/>
    <limit effort="1000" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="gripper_finger1_joint"/>
  </joint>
  <link name="gripper_finger1_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyx="-0.000001" iyy="0.000001" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger1_knuckle_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <link name="gripper_finger2_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.018491"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000009" ixy="-0.000001" ixz="0.000000" iyx="-0.000001" iyy="0.000001" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000010"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger2_knuckle_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <joint name="gripper_finger1_finger_joint" type="fixed">
    <parent link="gripper_finger1_knuckle_link"/>
    <child link="gripper_finger1_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <joint name="gripper_finger2_finger_joint" type="fixed">
    <parent link="gripper_finger2_knuckle_link"/>
    <child link="gripper_finger2_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.00408552455 -0.03148604435 0.0"/>
  </joint>
  <link name="gripper_finger1_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_finger_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyx="-0.000002" iyy="0.000021" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger1_finger_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <link name="gripper_finger2_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_finger_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_finger_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.027309"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000003" ixy="-0.000002" ixz="0.000000" iyx="-0.000002" iyy="0.000021" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000020"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger2_finger_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <joint name="gripper_finger1_inner_knuckle_joint" type="continuous">
    <parent link="gripper_base_link"/>
    <child link="gripper_finger1_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159265359 0.0 0.0" xyz="0.06142 0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="gripper_finger1_joint" offset="0"/>
  </joint>
  <joint name="gripper_finger2_inner_knuckle_joint" type="continuous">
    <parent link="gripper_base_link"/>
    <child link="gripper_finger2_inner_knuckle_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.06142 -0.0127 0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="gripper_finger1_joint" offset="0"/>
  </joint>
  <link name="gripper_finger1_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger1_inner_knuckle_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <link name="gripper_finger2_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_inner_knuckle_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_inner_knuckle_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.029951"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000039" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000035"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger2_inner_knuckle_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <joint name="gripper_finger1_finger_tip_joint" type="continuous">
    <parent link="gripper_finger1_inner_knuckle_link"/>
    <child link="gripper_finger1_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821 0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="gripper_finger1_joint" multiplier="-1"/>
  </joint>
  <joint name="gripper_finger2_finger_tip_joint" type="continuous">
    <parent link="gripper_finger2_inner_knuckle_link"/>
    <child link="gripper_finger2_finger_tip_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.04303959807 -0.03759940821  0.0"/>
    <limit effort="0.1" lower="-3.14" upper="3.14" velocity="100.0"/>
    <mimic joint="gripper_finger1_joint" multiplier="-1"/>
  </joint>
  <link name="gripper_finger1_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000006"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger1_finger_tip_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <link name="gripper_finger2_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/visual/robotiq_85_finger_tip_link.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_85_description/meshes/collision/robotiq_85_finger_tip_link.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.019555"/>
      <origin xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000002" ixy="0.000000" ixz="0.000000" iyx="0.000000" iyy="0.000005" iyz="0.000000" izx="0.000000" izy="0.000000" izz="0.000006"/>
    </inertial>
  </link>
  <gazebo reference="gripper_finger2_finger_tip_link">
    <implicitSpringDamper>1</implicitSpringDamper>
    <mu1>100000</mu1>
    <mu2>100000</mu2>
    <kp>100000000.0</kp>
    <kd>1.0</kd>
  </gazebo>
  <transmission name="gripper_finger1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_finger1_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="gripper_finger1_motor">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </actuator>
  </transmission>
  <gazebo>
    <!-- loading plugin: mimic joints works in gazebo now -->
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_1">
      <joint>gripper_finger1_joint</joint>
      <mimicJoint>gripper_finger2_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_2">
      <joint>gripper_finger1_joint</joint>
      <mimicJoint>gripper_finger2_inner_knuckle_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_3">
      <joint>gripper_finger1_joint</joint>
      <mimicJoint>gripper_finger1_inner_knuckle_joint</mimicJoint>
      <!--multiplier>1.0</multiplier>
                <offset>0.0</offset-->
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_4">
      <joint>gripper_finger1_joint</joint>
      <mimicJoint>gripper_finger1_finger_tip_joint</mimicJoint>
      <multiplier>-1.0</multiplier>
      <offset>0.0</offset>
    </plugin>
    <plugin filename="libroboticsgroup_gazebo_mimic_joint_plugin.so" name="mimic_robotiq_85_5">
      <joint>gripper_finger1_joint</joint>
      <mimicJoint>gripper_finger2_finger_tip_joint</mimicJoint>
      <multiplier>-1.0</multiplier>
      <offset>0.0</offset>
    </plugin>
  </gazebo>
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.163"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-6.28318530718" upper="6.28318530718" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.138 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-6.28318530718" upper="6.28318530718" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28"/>
      <inertia ixx="0.22689067591" ixy="0.0" ixz="0.0" iyy="0.22689067591" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.131 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
      <inertia ixx="0.049443313556" ixy="0.0" ixz="0.0" iyy="0.049443313556" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.392"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.127 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0171364731454" ixy="0.0" ixz="0.0" iyy="0.0171364731454" iyz="0.0" izz="0.033822"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.1 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="shoulder_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="upper_arm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="forearm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_1_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_3_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="wrist_2_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="ee_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="tool0"/>
  <joint name="wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.1 0"/>
    <parent link="wrist_3_link"/>
    <child link="tool0"/>
  </joint>
  <link name="ur_world"/>
  <joint name="world_joint" type="fixed">
    <parent link="ur_world"/>
    <child link="table"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
  <joint name="table_joint" type="fixed">
    <parent link="table"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.01 0.377 0.808"/>
  </joint>
  <joint name="UR5e_gripper" type="fixed">
    <parent link="tool0"/>
    <child link="gripper_root_link"/>
    <origin rpy="0.0 -1.570796327 0.0" xyz="0 0 0"/>
  </joint>
  <!--  <joint name="world_joint" type="fixed">-->
  <!--    <parent link="ur_world" />-->
  <!--    <child link = "base_link" />-->
  <!--    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />-->
  <!--  </joint>-->
</robot>
